packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -140,20 +134,64 @@ class _AddNewProductPageState extends State<AddNewProductPage>
       AnalyticsProductTracker(
         analyticsEvent: AnalyticsEvent.imagesNewProductPage,
         barcode: barcode,
-        check: () =>
-            _otherCount > 0 || _helper.isOneMainImagePopulated(upToDateProduct),
+        check: () => _otherCount > 0 || _helper.isOneMainImagePopulated(upToDateProduct),
       ),
     ];
     _daoProductList = DaoProductList(localDatabase);
     AnalyticsHelper.trackEvent(
       widget.events[EditProductAction.openPage]!,
       barcode: barcode,
     );
+    _totalPages = 3 + (widget.displayMisc ? 1 : 0) + (widget.displayPictures ? 1 : 0);
+    _progress = 1 / _totalPages;
+
+    _pageController.addListener(() {
+      setState(() {
+        _progress = (_pageController.page!.round() + 1) / _totalPages;
+        _isLastPage = (_pageController.page!.round() + 1) == _totalPages;
+      });
+    });
+  }
+
+  Future<bool> _onWillPop() async {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    if (_isPopulated) {
+      return true;
+    }
+    final bool? leaveThePage = await showDialog<bool>(
+      context: context,
+      builder: (final BuildContext context) => SmoothAlertDialog(
+        title: appLocalizations.new_product,
+        actionsAxis: Axis.vertical,
+        body: Text(appLocalizations.new_product_leave_message),
+        positiveAction: SmoothActionButton(
+          text: appLocalizations.yes,
+          onPressed: () => Navigator.of(context).pop(true),
+        ),
+        negativeAction: SmoothActionButton(
+          text: appLocalizations.cancel,
+          onPressed: () => Navigator.of(context).pop(false),
+        ),
+      ),
+    );
+    if (leaveThePage == true) {
+      AnalyticsHelper.trackEvent(
+        widget.events[EditProductAction.leaveEmpty]!,
+        barcode: barcode,
+      );
+    }
+    return leaveThePage ?? false;
+  }
+
+  @override
+  void dispose() {
+    _pageController.dispose();
+    super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    _colorScheme = _colorScheme = Theme.of(context).colorScheme;
```

Comment:
Perfect, please just remove the attribute from the class, as it's not used anymore